lib/websocket.go

```diff
@@ -16,11 +12,25 @@ import (
 
 const qriWebsocketProtocol = "qri-websocket"
 
-// ServeWebsocket creates a websocket that clients can connect to in order to
-// get realtime events
-func (inst *Instance) ServeWebsocket(ctx context.Context) {
-	apiCfg := inst.cfg.API
+// Collect all websocket connections
+var wsConnections = []*websocket.Conn{}
```

Comment:
Not a fan of having this as a global, especially since we'll need to do dead connection mgmt for long-lived processes. Let's do an interface with a single `Serve` method & + private implementation:

```go
type WebsocketHandler interface {
  WSConnectionHandler(w http.ResponseWriter, r *http.Request)
}

type wsHandler struct {
  conns []*websocket.Conn
}

var _ WebsocketHandler = (*wsHandler)(nil)

func NewWebsocketHandler(ctx context.Context, bust event.Bus, r repo.Repo) WebsocketHandler {
 // ...
}

func (h *wsHandler)  WSConnectionHandler(w http.ResponseWriter, r *http.Request) {
  // ...
}
```

While we're here, looks like the constructor doesn't need a full instance, just `event.Bus` & `repo.Repo` args.